dev/docker-compose.yaml

```diff
@@ -14,14 +14,14 @@ services:
       - internal
 
   fhir:
-    image: hapiproject/hapi:${FHIR_IMAGE_TAG:-v6.6.0}
+    image: hapiproject/hapi:${FHIR_IMAGE_TAG:-v7.0.3}
     environment:
       SPRING_CONFIG_LOCATION: file:///opt/application.yaml
       spring.datasource.url: jdbc:postgresql://db:5432/hapifhir
       spring.datasource.username: postgres
       spring.datasource.password: postgres
       spring.datasource.driverClassName: org.postgresql.Driver
-      spring.jpa.hibernate.dialect: org.hibernate.dialect.PostgreSQL94Dialect
+      spring.jpa.properties.hibernate.dialect: ca.uhn.fhir.jpa.model.dialect.HapiFhirPostgres94Dialect
 
```

Comment:
I'd suggest configuring any overrides to the [default application.yaml](https://github.com/hapifhir/hapi-fhir-jpaserver-starter/blob/image/v7.0.3/src/main/resources/application.yaml) as environment variables (and use the [vanilla application.yaml from Hapi](https://github.com/hapifhir/hapi-fhir-jpaserver-starter/blob/image/v7.0.3/src/main/resources/application.yaml))

Since env vars take precedence over values in application.yaml, it may not be obvious later which config is actually being used. It'll also be fewer changes to maintain when Hapi is upgraded in the future.

In the past, Hapi has required a application.yaml, but ideally we'd remove that in the future (since it's all defaults anyway)
```suggestion

      # enable IPS; use latest version
      hapi.fhir.ips_enabled: "true"
      hapi.fhir.implementationguides.ips_1_1_0.packageUrl: https://build.fhir.org/ig/HL7/fhir-ips/package.tgz
      hapi.fhir.implementationguides.ips_1_1_0.name: hl7.fhir.uv.ips
      hapi.fhir.implementationguides.ips_1_1_0.version: 1.1.0

      # IPS needs this for $REASON
      hapi.fhir.allow_external_references: "true"
      hapi.fhir.enforce_referential_integrity_on_write: "false"
      hapi.fhir.auto_create_placeholder_reference_targets: "true"
      hapi.fhir.cr_enabled: "true"
```
(I [tested the above values via `/actuator/configprops`](https://fhir.ips-dev.fl.ivanc.dev.cirg.uw.edu/actuator/configprops) to confirm they work)